### PR TITLE
Feature/cx 5306 customisation links

### PR DESCRIPTION
**Pull request by kopijunkie**
# Task
We want to offer customers supported customisation options for the JS SDK so that they can customise the underline and hover, active state background colour of the link elements. Note that some of the "links" are really text buttons.
Screenshot 2021-02-23 at 10 12 20
Screenshot 2021-02-02 at 11 33 32
Screenshot 2021-02-03 at 16 05 19

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [x] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

File: README.md [hunk 1]
```diff
@@ -424,6 +424,24 @@ A number of options are available to allow you to customise the SDK:
...
+  | Warning, Error Popups              | Description                                                |
+  | ---------------------------------- | ---------------------------------------------------------- |
+  | `colorContentAlertInfo`            | Change warning popup fallback Link underline color         |
+  | `colorBackgroundAlertInfo`         | Change warning popup fallback Link background on hover     |
+  | `colorContentAlertInfoLinkHover`   | Change warning popup fallback Link background on click/tap |
+  | `colorContentAlertInfoLinkActive`  | Change warning popup fallback Link background on click/tap |
+  | `colorContentAlertError`           | Change error popup fallback Link underline color           |
+  | `colorBackgroundAlertError`        | Change error popup fallback Link background on hover       |
+  | `colorContentAlertErrorLinkHover`  | Change error popup fallback Link background on click/tap   |
+  | `colorContentAlertErrorLinkActive` | Change error popup fallback Link background on click/tap   |
```
Comment (eraxyz) [inline README.md:443]: I think there may been some copy/paste errors here? The two options with `Hover` in the name mention `click/tap` in the description and there are two options with `hover` in the description but not in the name.

Also `Content` in the name of some but `background` in the description.

File: src/components/Theme/style.scss [hunk 2]
```diff
@@ -257,70 +257,76 @@ $logo-height: 32 * $unit;
...
 
-.warning {
+.link:visited {
+  text-decoration: none;
+  color: $color-body-text;
+}
+
+/* Link/Text Button for Live Doc/Face Capture fallback on Warning popup */
+.warningFallbackButton {
+  color: --osdk-color-content-alert-info;
```
Comment (eraxyz) [inline src/components/Theme/style.scss:298]:
```suggestion
  color: var(--osdk-color-content-alert-info);
```

**Review by eraxyz — CHANGES_REQUESTED** [review]
Looks good for the most part.

I just have a comment about the use of castor tokens - if `castor.color('content-negative')` and `rgba(var(--ods-color-content-negative))` are the same then we should probably choose a syntax and use it throughout. I didn't even know `castor.color()` was a thing so not sure if they are exactly the same.

`castor.color` used in:
src/components/Uploader/style.scss
src/components/Theme/style.scss

`--ods` used in:
src/components/crossDevice/CrossDeviceLink/style.scss
src/components/Theme/tokens.scss

File: README.md (hunk 1 above)
Comment (kopijunkie) [reply README.md:470]: Oops. Thanks for catching this! Thought I cleaned it up, but guess not. 🙈

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by kopijunkie** [comment]
> I just have a comment about the use of castor tokens - if `castor.color('content-negative')` and `rgba(var(--ods-color-content-negative))` are the same then we should probably choose a syntax and use it throughout. I didn't even know `castor.color()` was a thing so not sure if they are exactly the same.

I tried using the Castor token helpers in `tokens.scss` after Castor squad asked me about why we're not using them, but it looks like somehow the colours don't get applied if the helpers are used there where it does in the component's own stylesheets. I'll just revert to using the token itself throughout for now.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

**Review by eraxyz — APPROVED** [review]

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Button/FallbackButton.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ import style from './style.scss'
 const FallbackButton = ({ text, onClick }) => (
   <button
     type="button"
-    className={`${style.fallbackButton} ${theme.warning}`}
+    className={`${style.fallbackButton} ${theme.warningFallbackButton}`}
```
Comment (ethan605) [inline src/components/Button/FallbackButton.js:8]: Just a minor change:

```suggestion
    className={classNames(style.fallbackButton, theme.warningFallbackButton)}
```

**Review by ethan605 — APPROVED** [review]
LGTM, just suggested some minor changes.

File: src/components/Capture/Face.js [hunk 4]
```diff
@@ -66,9 +67,9 @@ class Face extends Component {
   handleFileSelected = (file) =>
     validateFile(file, this.handleUpload, this.handleError)
 
-  renderUploadFallback = (text) => (
+  renderUploadFallback = (text, type = 'warning') => (
     <CustomFileInput
-      className={style.uploadFallback}
+      className={theme[`${type}FallbackButton`]}
```
[the remark this answers is not in the document]
Comment (kopijunkie) [reply src/components/Capture/Face.js:72]: Thanks. Yeah, agree that this is actually better/safer on thinking it through more.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1340-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1340-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**